lxd/util/sys.go

```diff
@@ -110,3 +110,16 @@ func AddFileDescriptor(fdFiles *[]*os.File, file *os.File) int {
 	*fdFiles = append(*fdFiles, file)
 	return 2 + len(*fdFiles) // Use 2+fdFiles count, as first user file descriptor is 3.
 }
+
+// ShortenedSocketPath creates a shorter alternative path to a socket by using the file descriptor to the directory of the socket file.
+// Used to handle paths > 108 chars.
+// Files opened here must be closed outside this function once they are not needed anymore.
+func ShortenedSocketPath(originalSockPath string, fdFiles *[]*os.File) (string, error) {
+	configSockDir, err := os.Open(filepath.Dir(originalSockPath))
+	if err != nil {
+		return "", fmt.Errorf("Failed to open device socket directory %q: %w", filepath.Dir(originalSockPath), err)
+	}
+
+	// Open a file descriptor to the directory where socket lies to avoid accessing the file descriptor to the socket directly.
```

Comment:
@mihalicyn please could you help us with why this approach is needed? Thanks

Reply:
@hamistao not addressed yet. cc @mihalicyn

Reply:
@hamistao please work with @mihalicyn to update this comment to explain why this approach is used in a follow up PR. Thanks

Reply:
I used `// Open a file descriptor to the directory where socket lies to avoid accessing the file descriptor to the sockfs inode.` for now since we decided to switch to the `O_PATH` approach.